ntp-proto/src/peer.rs

```diff
@@ -237,12 +237,7 @@ impl Peer {
         // the origin_timestamp changed; the server is not allowed to modify this field
         let origin_changed = Some(message.origin_timestamp) != self.next_expected_origin;
 
-        if message.mode != NtpAssociationMode::Server {
-            // we currently only support a client <-> server association
-            Err(IgnoreReason::InvalidMode)
-        } else if transmit_unchanged || origin_changed {
-            Err(IgnoreReason::InvalidPacketTime)
-        } else if message.is_kiss_rate() {
+        if message.is_kiss_rate() {
```

Comment:
messages with a kiss code make no guarantees about the other fields, but we still must respect the kiss codes. So, check the kiss codes first